packages/blade/src/utils/isReactNative/platform.all.ts

```diff
@@ -0,0 +1,32 @@
+// THIS .all file is special only for platform.ts file. Do not use anywhere else
+// This file only contains type level code and putting any runtime code will cause errors since our
+// bundler is not setup to handle this extension.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-namespace */
+import type { Brand, IsFunction, NativeOrWebBrand, _brand } from './types';
+
+namespace Platform {
+  export type Name = 'web' | 'native';
+  // Unionify both web & native types, this will help us blade developers internally
+  export type Select<Options extends { web: any; native: any }> = IsFunction<
+    Options['native']
+  > extends true
+    ? IsFunction<Options['web']> extends true
+      ? // functions has to be intersected
+        Brand<Options['native'] & Options['web'], 'platform-all'>
+      : Brand<Options['native'], 'platform-native'> | Brand<Options['web'], 'platform-web'>
+    : Brand<Options['native'], 'platform-native'> | Brand<Options['web'], 'platform-web'>;
```

Comment:
Suggestion: can we simplify this? Break into multiple intermediate types?

Reply:
Will try.

Reply:
I've simplified it a bit, let me know if it's more readable. 

```ts
  // Unionify both web & native types, this will help us blade developers internally
  export type Select<Options extends { web: any; native: any }> = IsBothFunction<
    Options['native'],
    Options['web']
  > extends true
    ? // functions has to be intersected
      Brand<Options['native'] & Options['web'], 'platform-all'>
    : Brand<Options['native'], 'platform-native'> | Brand<Options['web'], 'platform-web'>;

```

Reply:
A bit better